subsys/net/ip/ipv4.h

```diff
@@ -52,4 +52,33 @@ struct net_pkt *net_ipv4_create(struct net_pkt *pkt,
  */
 void net_ipv4_finalize(struct net_pkt *pkt, u8_t next_header_proto);
 
+
+/**
+ * @brief Create IPv4 packet in provided net_pkt.
+ *
+ * @param pkt Network packet
+ * @param src Source IPv4 address
+ * @param dst Destination IPv4 address
+ * @param next_header Protocol type of the next header after IPv4 header.
+ *
+ * @return 0 on success, negative errno otherwise.
+ */
+int net_ipv4_create_new(struct net_pkt *pkt,
+			const struct in_addr *src,
+			const struct in_addr *dst,
+			u8_t next_header_proto);
```

Comment:
Why do we need to have next_header_proto parameter here and in finalize function, it looks like one should be enough.

Reply:
you mean we would inform ipv4 header about the protocol only at finalize? why not indeed.